### PR TITLE
Adds theme

### LINKED ISSUE #5: Have themes
It would be nice if user could choose from different themes that change colors and decoration etc.

### PR DESCRIPTION
**Pull request by astrale-sharp**
Closes #5

### CODE REVIEW

File: slides.typ [hunk 1]
```diff
@@ -111,54 +112,32 @@
     }
 
     set text(
-        size: 25pt,
+        size: theme.text-size,
+    )
+    let this-slide = current-slide(
+        section: section,
+        logical-slide : logical-slide,
+        subslide : subslide,
```
Comment (PgBiel) [inline slides.typ:120]: Nitpick
```suggestion
        logical-slide: logical-slide,
        subslide: subslide,
```

File: slides.typ [hunk 2]
```diff
@@ -92,14 +94,13 @@
...
 
 #let slides(
-    title: none,
-    author: none,
-    short-title: none,
-    short-author: none,
-    date: none,
-    color: teal,
+    data : none,
+    theme : base_theme(),
```
Comment (PgBiel) [inline slides.typ:98]: Nitpick
```suggestion
    data: none,
    theme: base_theme(),
```

File: slides.typ [hunk 3]
```diff
@@ -16,7 +18,7 @@
     )
 }
 
-#let slide(max-repetitions: 10, body) = {
+#let slide(theme: base_theme(),max-repetitions : 10, body) = {
```
Comment (PgBiel) [inline slides.typ:21]: Nitpick
```suggestion
#let slide(theme: base_theme(), max-repetitions: 10, body) = {
```

File: examples/simple.typ [hunk 4]
```diff
@@ -1,12 +1,16 @@
 #import "../slides.typ": *
 
 #show: slides.with(
+  data : data(
```
Comment (PgBiel) [inline examples/simple.typ:4]: Nitpick
```suggestion
  data: data(
```

File: theme.typ [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            )}
+    
+
+#let data(
+    author : none,
+    title : none,
+    short-author : none, 
+    short-title : none, 
+    date : none, 
+)= {
```
Comment (PgBiel) [inline theme.typ:22]: Nitpick
```suggestion
) = {
```

**Review by PgBiel — CHANGES_REQUESTED** [review]
Hey, thanks for the contribution! I've pointed out several formatting nitpicks - there are probably more instances of them, so make sure to fix those too (but please note that this isn't in any way about the quality of the code, just what I think would be better formatting):

File: slides.typ [hunk 6]
```diff
@@ -111,54 +112,32 @@
...
-                    #v(1em)
-                    #date
+                    #text(theme.title-text-size)[*#data.title*] \
+                    #v(theme.title-text-space)
+                    #text(theme.author-text-space)[#data.author] \
+                    #v(theme.title-text-space)
+                    #data.date
                 ]
             )
         )
```
Comment (PgBiel) [inline slides.typ:143]: Worth discussing this further, but I just wanted to comment that we should probably let themes entirely customize the title slide as well (return a content or similar)

Should also be able to use lambdas for this (with the "big state dict" we discussed previously)

**Comment by astrale-sharp** [comment]
Hope I havent forgotten any fmt nitpicks, i went a bit fast !

**Comment by andreasKroepelin** [comment]
Hey, thanks a lot for your contribution! I'm a bit sick at the moment so it will take me a few days until I can properly review this PR. Just as a quick question from scrolling through the code: Is my impression right that you intend to set the theme for every individual slide? That would seem like an rather unfortunate API to me (you also don't do that in LaTeX beamer). I think specifying the theme should happen as an argument to the `#slides`-template function, that initiates the document. Sorry if I just got it wrong, though.

**Comment by astrale-sharp** [comment]
Take care and rest well!! :heart: 
Indeed I'm not happy about specifying the theme each time..
I will add a state for theme !

**Comment by astrale-sharp** [comment]
The new commit introduces a state for theme but introduces a bug (why? :() where first page is blank

**Comment by enseignantePC** [comment]
The bug was due to a locate call
`#let _ = locate(....)` fixed it

**Pull request closed without merging**